ndmg/scripts/ndmg_bids.py

```diff
@@ -229,36 +229,42 @@ def session_level(
     # optional args stored in kwargs
     # use worker wrapper to call function f with args arg
     # and keyword args kwargs
-    print(args)
-    ndmg_dwi_worker(
-        args[0][0],
-        args[0][1],
-        args[0][2],
-        args[0][3],
-        atlas,
-        atlas_mask,
-        labels,
-        outDir,
-        vox_size,
-        mod_type,
-        track_type,
-        mod_func,
-        seeds,
-        reg_style,
-        clean,
-        skipeddy,
-        skipreg,
-        buck=buck,
-        remo=remo,
-        push=push,
-        creds=creds,
-        debug=debug,
-        modif=modif,
-    )
-    rmflds = []
-    if len(rmflds) > 0:
-        cmd = "rm -rf {}".format(" ".join(rmflds))
-        mgu.execute_cmd(cmd)
+    out_Dir = outDir
+    for x in range(len(args)):
+        print(args)
+
+        if len(args) > 1:
+            bid_len = len(dwis[x].split("/"))
+            outDir = out_Dir+"/ndmg-batch/"+dwis[x].split("/")[bid_len-4]+"/"+dwis[x].split("/")[bid_len-3]
```

Comment:
Hey, your first bit of Python code, nice!

so, for `args`:
`args[0]`: List == the input data for the first session + MNI + atlas
`args[1]`: List == the input data for the second session + MNI + atlas
and so on

```
for x in range(len(args)):
    print(args)
```
will print all of the args len(args) times. I don't think that's what we want.
If you're trying to print all of the arguments for each set of input data, you can just say:

```
for arg in args:
    print(arg)
```
Python lets you iterate directly through the objects in iterables without using a number index (which is different than C++, I think?)

The OutDir line is pretty messy. I'd look into `f-string` formatting: https://realpython.com/python-f-strings/

Reply:
Also, I don't understand where this code loops through subjects/sessions if you're trying to run a dataset locally.

Looping through on s3 isn't really super necessary, since every time we use s3, we'll just use AWS Batch for it
So it should be a loop on `ndmg_dwi_worker`, yeah?

Reply:
Ok, I understand what this is doing now that I saw that `ndmg_dwi_worker` is part of this loop, which I didn't see the first time around

I still think it's generally more pythonic to do `for arg in args` rather than `for x in range(len(args))`, although since this works, we could keep it indexing by number rather than object

If we want to iterate by index, I'd look into `[enumerate](http://docs.quantifiedcode.com/python-anti-patterns/readability/using_an_unpythonic_loop.html?highlight=enumerate)`

Reply:
Let's not make a `ndmg-batch` intermediate folder as well - the output directory should just look like `output/sub-x/ses-y/...`

Reply:
Yeah, that is better form to change it to `enumerate`, which I'll look into. As for the `ndmg-batch' intermediate folder, I wanted to keep some folder between /output and /sub-x/ to prevent accidental overwriting of other outputs but that's a fringe case where they could just specify an different output folder

Reply:
For the OutDir line, I'd also look into `pathlib`, which is python3's native library for working with filepaths: 
https://realpython.com/python-pathlib/
https://docs.python.org/3/library/pathlib.html

Reply:
Fixed it, turns out I was super over-complicating things, without the addition of the /batch/ intermediate directory arg[7] contains the proper output directory